src/util/tubemap.js

```diff
@@ -3109,26 +3116,129 @@ function drawNodes(dNodes) {
+    "Aligned Reads:",
+    currentNode.incomingReads.length +
+      currentNode.internalReads.length +
+      currentNode.outgoingReads.length,
+  ]);
+  nodeAttributes.push([
+    "Number of Reads Visiting Node:",
+    numReadsVisitNode(currentNode),
+  ]);
+  nodeAttributes.push(["Coverage:", coverage(currentNode, reads)]);
+
+  console.log("Single Click");
+  console.log("node show info callback", config.showInfoCallback);
+  config.showInfoCallback(nodeAttributes);
+}
+
+export function numReadsVisitNode(node) {
+  let countReads = new Set();
+  // incoming reads are reads that enter the node but don't start within it. They are represented as
+  // an array of subarrays which have 2 elements: an index indicating the read index and read path's index.
+  // The first node will not have any incoming reads.
+  for (let readVisit of node.incomingReads) {
+    countReads.add(readVisit[0]);
+  }
+  // internal reads are reads that start and end within the node. They are represented as
+  // an array of values which indicate the read index.
+  for (let read of node.internalReads) {
+    countReads.add(read);
+  }
+  // outgoing reads are reads that exit the node when the read starts within it. They are represented as
+  // an array of subarrays which have 2 elements: an index indicating the read index and read path's index.
+  // The last node will not have any outgoing reads.
+  for (let readVisit of node.outgoingReads) {
+    countReads.add(readVisit[0]);
+  }
+  return countReads.size;
+}
+
+// computes average number of reads passing through each base in the node
+export function coverage(node, allReads) {
+  if (node.sequenceLength === 0) {
+    return 0.0;
+  }
+  let countBases = 0;
+  for (let readVisit of node.incomingReads) {
+    let readNum = readVisit[0];
+    let readPathIndex = readVisit[1];
+    let currRead = allReads[readNum];
+    let numNodes = currRead.sequenceNew.length;
+    //  if current node is the last node on the read path, add the finalNodeCoverLength number of bases
+    if (numNodes === readPathIndex + 1) {
+      countBases += currRead.finalNodeCoverLength;
+      // otherwise add the node's sequence length (width of node in bases)
+    } else {
+      countBases += node.sequenceLength;
+    }
+  }
+  // internal reads
+  for (let readVisit of node.internalReads) {
+    // compute coverage of read by finalNodeCoverLength and firstNodeOffset fields
+    //  indicating read's starting and ending points within the node.
+    let readNum = readVisit;
+    let currRead = allReads[readNum];
+    countBases += currRead.finalNodeCoverLength - currRead.firstNodeOffset;
+  }
+  // outgoing reads
+  for (let readVisit of node.outgoingReads) {
+    let readNum = readVisit[0];
+    let currRead = allReads[readNum];
+    // coverage of outgoing read would be the the distance between the end of the node and the
+    //  starting point of the read within the node
+    countBases += node.sequenceLength - currRead.firstNodeOffset;
```

Comment:
I think we're forgetting to skip the read here when this read is also an `incomingReads` read. At this line we add the length of this node minus the offset along the first node. That only makes sense if this *is* the first node on the read, and the read is not also incoming, but we don't seem to check for that.

Reply:
We don't need to do this. Incoming and outgoing reads are in fact disjoint; incoming reads also includes reads that make it through the node.